.github/workflows/build_and_test.yml

```diff
@@ -66,12 +68,20 @@ jobs:
       if: matrix.os == 'macos-latest'
       run: echo "OPENSSL_ROOT_DIR=$(brew --prefix openssl@1.1)" >> $GITHUB_ENV
 
+    - name: Set OpenSSL location (Linux)
+      if: matrix.os == 'ubuntu-latest'
+      run: echo "RUSTFLAGS=-L /usr/lib/x86_64-linux-gnu" >> $GITHUB_ENV
```

Comment:
This is a bit of a hack, but we can't use `OPENSSL_ROOT_DIR` for Linux since `paho-mqtt-sys` is hardcoded to look in `$OPENSSL_ROOT_DIR/lib`. Ubuntu allows multiple architectures, so the `amd64` version of `libssl` is located in `/usr/lib/x86_64-linux-gnu`